Samples/DirectMLNpuInference/main.cpp

```diff
@@ -30,36 +65,39 @@ void InitializeDirectML(ID3D12Device1** d3dDeviceOut, ID3D12CommandQueue** comma
             dxcoreCreateAdapterFactory(IID_PPV_ARGS(&factory));
         }
     }
-    // Create the DXCore Adapter
+
+    // Create the DXCore Adapter, for the purposes of selecting NPU we look for (!GRAPHICS && (GENERIC_ML || CORE_COMPUTE))
     ComPtr<IDXCoreAdapter> adapter;
+    ComPtr<IDXCoreAdapterList> adapterList;
     if (factory)
     {
-        const GUID dxGUIDs[] = { DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE };
-        ComPtr<IDXCoreAdapterList> adapterList;
-        THROW_IF_FAILED(factory->CreateAdapterList(ARRAYSIZE(dxGUIDs), dxGUIDs, IID_PPV_ARGS(&adapterList)));
-        for (uint32_t i = 0, adapterCount = adapterList->GetAdapterCount(); i < adapterCount; i++)
+        THROW_IF_FAILED(factory->CreateAdapterList(1, &DXCORE_ADAPTER_ATTRIBUTE_D3D12_GENERIC_ML, IID_PPV_ARGS(&adapterList)));
+
+        if (adapterList->GetAdapterCount() > 0)
+        {
+            GetNonGraphicsAdapter(adapterList.Get(), adapter.GetAddressOf());
```

Comment:
See comment about removing output parameter, but using `adapter.GetAddressOf()` will leak. You'd need `commandList.ReleaseAndGetAddressOf()` to not leak here.